core/distributed/helpers.hpp

```diff
@@ -122,11 +122,15 @@ void vector_dispatch(T* linop, F&& f, Args&&... args)
 {
 #if GINKGO_BUILD_MPI
     if (is_distributed(linop)) {
-        using type = std::conditional_t<
-            std::is_const<T>::value,
-            const experimental::distributed::Vector<ValueType>,
-            experimental::distributed::Vector<ValueType>>;
-        f(dynamic_cast<type*>(linop), std::forward<Args>(args)...);
+        if constexpr (std::is_same_v<remove_complex<ValueType>, half>) {
```

Comment:
Is there a reason we dont support half with distributed ? Or will it come in a future PR ?

Reply:
half precision transfer is trivial but the reduce operation and one side accumulation is not trivial.
I have tried adding the custom `MPI_Op` to support reduce on custom type, which works.
However, one side accumulation does not allow any non-predefined operation.
We will only support half in this case.
Or, use relatively new compiler and MPI standard with its extension, so they can support half natively.
It will be future PR anyways. It might not be in this release, either.